src/FileSystem/FileSystem.cpp

```diff
@@ -559,13 +560,11 @@ bool CFileSystem::extract(const std::string& filename,
 			delete archive;
 			return false;
 		}
-		int res = 1;
-		if (!buf.empty())
```

Comment:
I wouldn't drop the emptiness check here, I can't find in the documentation any guarantees on what fwrite does with this pointer if count == 0.